crates/sncast/tests/e2e/script/logging.rs

```diff
@@ -0,0 +1,200 @@
+        "dummy",
+        "contracts/src/lib.cairo",
+        salt,
+    )
+}
+
+#[tokio::test]
+async fn test_default_verbosity() {
+    let current_dir = duplicate_map_script("13");
+    let accounts_json_path = get_accounts_path("tests/data/accounts/accounts.json");
+
+    let script_name = "map_script_all_cheatcodes";
+    let args = vec![
+        "--accounts-file",
+        accounts_json_path.as_str(),
+        "--account",
+        "user2",
+        "--url",
+        URL,
+        "script",
+        &script_name,
+    ];
+
+    let snapbox = Command::new(cargo_bin!("sncast"))
+        .current_dir(current_dir.path().join("scripts"))
+        .args(args);
+
+    snapbox.assert().success().stdout_matches(indoc! {r"
+           Compiling lib(map_script) map_script v0.1.0 [..]
+           Compiling starknet-contract(map_script) map_script v0.1.0 [..]
+            Finished release target(s) in [..] seconds
+        script_subcommand: get_nonce
+        nonce: [..]
+
+        script_subcommand: declare
+        class_hash: 0x[..]
+        transaction_hash: 0x[..]
+
+        script_subcommand: get_nonce
+        nonce: [..]
+
+        script_subcommand: deploy
+        contract_address: 0x[..]
+        transaction_hash: 0x[..]
+
+        script_subcommand: get_nonce
+        nonce: [..]
+
+        script_subcommand: invoke
+        transaction_hash: 0x[..]
+
+        script_subcommand: call
+        response: [0x2]
+
+        command: script
+        status: success
+    "});
+}
+
+#[tokio::test]
+async fn test_default_verbosity_with_json() {
+    let current_dir = duplicate_map_script("14");
+    let accounts_json_path = get_accounts_path("tests/data/accounts/accounts.json");
+
+    let script_name = "map_script_all_cheatcodes";
+    let args = vec![
+        "--accounts-file",
+        accounts_json_path.as_str(),
+        "--account",
+        "user3",
+        "--url",
+        URL,
+        "--json",
+        "script",
+        &script_name,
+    ];
+
+    let snapbox = Command::new(cargo_bin!("sncast"))
+        .current_dir(current_dir.path().join("scripts"))
+        .args(args);
+
+    snapbox.assert().success().stdout_matches(indoc! {r#"
+        {"status":"compiling","message":"lib(map_script) map_script v0.1.0 ([..])"}
+        {"status":"compiling","message":"starknet-contract(map_script) map_script v0.1.0 ([..])"}
+        {"status":"finished","message":"release target(s) in [..] seconds"}
+        {"script_subcommand":"get_nonce","nonce":"[..]"}
+
+        {"script_subcommand":"declare","class_hash":"0x[..]","transaction_hash":"0x[..]"}
+
+        {"script_subcommand":"get_nonce","nonce":"[..]"}
+
+        {"script_subcommand":"deploy","contract_address":"0x[..]","transaction_hash":"0x[..]"}
+
+        {"script_subcommand":"get_nonce","nonce":"[..]"}
+
+        {"script_subcommand":"invoke","transaction_hash":"0x[..]"}
+
+        {"script_subcommand":"call","response":["0x2"]}
+
+        {"command":"script","status":"success"}
```

Comment:
this output is not useful for anything at the moment, as this will yield errors for any tool that want's to parse it (this is not a valid json). We can either put everything in a list (not very useful, as items probably won't hold their position, so we'd need additional timestamp or something) or (better) the whole output should be in one json object. 

Let's create an issue and try to fix this problem in a separate PR.

Reply:
this was inspired by scarb - first three lines is the output from scarb

Reply:
although I added empty lines to the script output, maybe it's better to remove them. The user can read the output line by line